File: app/src/main/java/com/saint/struct/tool/CrashHandler.java
package com.saint.struct.tool;

import android.util.Log;

public class CrashHandler implements Thread.UncaughtExceptionHandler {
    @Override
    public void uncaughtException(Thread thread, Throwable ex) {
        Log.e("CrashHandler","CrashHandler = " + System.currentTimeMillis());
    }
}

File: app/src/main/java/com/saint/struct/tool/CrashHandler.kt
package com.saint.struct.tool

import android.os.Looper
import android.util.Log

class CrashHandler : Thread.UncaughtExceptionHandler {
    override fun uncaughtException(thread: Thread, ex: Throwable) {
        Log.e("CrashHandler", "CrashHandler = " + System.currentTimeMillis())
        try {
            //fix No Looper; Looper.prepare() wasn't called on this thread.
            if (Looper.myLooper() == null) {
                Looper.prepare();
            }
            Looper.loop();
        } catch (e: Exception) {
            uncaughtException(Thread.currentThread(), e);
        }
    }
}
